adrs/2023-02-10-limit-manager-role-permission.md

```diff
@@ -66,18 +66,54 @@ To help these customers and improve security for `actions-runner-controller` in
 - List/Watch on `RoleBindings`
 - List/Watch on `ServiceAccounts`
 
-> We will change the default cache-based client to bypass cache on reading `Secrets`, so we can eliminate the need for `List` and `Watch` `Secrets` permission in cluster scope.
+> We will change the default cache-based client to bypass cache on reading `Secrets` and `ConfigMaps`(ConfigMap is used when you configure `githubServerTLS`), so we can eliminate the need for `List` and `Watch` `Secrets` permission in cluster scope.
 
-Introduce a new `Role` per `AutoScalingRunnerSet` installation and `RoleBinding` the `Role` with the controller's service account in the namespace that each `AutoScalingRunnerSet` deployed with the following permission.
+Introduce a new `Role` for the controller and `RoleBinding` the `Role` with the controller's service account in the namespace the controller is deployed, this role will grant the controller service account required permission to work with `AutoScalingListeners` in the controller namespace.
 
+- Get/Create/Delete on `Pods`
+- Get on `Pods/status`
 - Get/Create/Delete/Update/Patch on `Secrets`
-- Get/Create/Delete/Update/Patch on `Pods`
+- Get/Create/Delete/Update/Patch on `ServiceAccounts`
+
+The `Role` and `RoleBinding` creation will happen during the `helm install demo oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set-controller`
+
+During `helm install demo oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set-controller`, we will store the controller's service account info as labels on the controller `Deployment`.
+Ex:
+```yaml
+    actions.github.com/controller-service-account-namespace: {{ .Release.Namespace }}
+    actions.github.com/controller-service-account-name: {{ include "gha-runner-scale-set-controller.serviceAccountName" . }}
+```
+
+Introduce a new `Role` per `AutoScalingRunnerSet` installation and `RoleBinding` the `Role` with the controller's service account in the namespace that each `AutoScalingRunnerSet` deployed with the following permission.
```

Comment:
```suggestion
Introduce a new `Role` per `AutoScalingRunnerSet` installation and `RoleBinding` the `Role` with the controller's `ServiceAccount` in the namespace that each `AutoScalingRunnerSet` deployed with the following permission.
```
nit: if you don't agree with the previous suggestion, feel free to ignore this one as well :relaxed: